File: armstrong.py
a = int(input("Enter your number: "))
b = a
sum = 0

while (a>0):
    c = a%10
    T = c*c*c
    sum = sum + T
    a = a//10

if (sum == b):
    print (b, "is an armstrong number")
    if (b%2!=0):
        print (b, "is also an odd number")
    else:
        print (b, "is an armstrong number but not an odd number")
else:
    print (b, "is not an armstrong number")

File: armstrong.c
#include<stdio.h>
void main(){
    int n,i=0, r,sum=0,temp;
    printf("please enter the number you want to check for  = ");
    scanf("%d",&n);
    temp=n;
    while (i<n)
    {
        r = n%10;
        sum = sum+(r*r*r);
        n=(n/10);
        

    }
    if (sum==temp){
        printf("number is armstrong.");

    }
    else{
        printf("not armstrong.");
    }
    

}